.cbmc-batch/jobs/Makefile.common

```diff
@@ -0,0 +1,211 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL=/bin/bash
+
+default: report
+
+# if Makefile.local exists, use it. This provides a way to override the defaults
+sinclude Makefile.local
+
+################################################################
+# Define locations and binaries
+ABSTRACTIONS ?= ""
+BASEDIR ?= $(abspath ../../../..)
+BATCH ?= cbmc-batch
+
+CBMCFLAGS += \
+	--bounds-check \
+	--pointer-check \
+	--function $(ENTRY) \
+	$(CBMC_UNWINDSET)
+
+GOTO_ANALYZER ?= goto-analyzer
+GOTO_CC ?= goto-cc
+GOTO_INSTRUMENT ?= goto-instrument
+HELPERDIR ?= $(SRCDIR)/.cbmc-batch
+
+INC += \
```

Comment:
does this work, given that srcdir is only defined later in the file?

Reply:
Yes. In the Makefile.common, the assignment `SRCDIR ?= $(abspath ../../..)` is equivalent to 

```
ifeq ($(origin SRCDIR), undefined)
SRCDIR = $(abspath ../../..)
endif
```

In that case, `SRCDIR` is declared as a recursively expanded variable, so 

> these references are expanded whenever this variable is substituted (in the course of expanding some other string)

 (http://www.chiark.greenend.org.uk/doc/make-doc/make.html/Using-Variables.html#Flavors)

When we run `make` in one of the harness and it goes to the first rule:

```
$(ENTRY)1.goto: $(ENTRY).c $(DEPENDENCIES)
	$(GOTO_CC) $< --function $(ENTRY) $(DEPENDENCIES) $(INC) -o $@ \
	        2>&1 | tee $(ENTRY)1.log
```

All dependencies are expanded with its absolute path, which in my case is:

```
goto-cc aws_array_list_init_dynamic_harness.c --function aws_array_list_init_dynamic_harness
/Users/_username_/Documents/projects/aws-c-common/source/array_list.c 
/Users/_username_/Documents/projects/aws-c-common/source/byte_buf.c 
/Users/_username_/Documents/projects/aws-c-common/source/common.c 
/Users/_username_/Documents/projects/aws-c-common/source/error.c
-I/Users/_username_/Documents/projects/aws-c-common/.cbmc-batch/.
-I/Users/_username_/Documents/projects/aws-c-common/include/
-o aws_array_list_init_dynamic_harness1.goto \
	        2>&1 | tee aws_array_list_init_dynamic_harness1.log
```